services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -0,0 +1,484 @@
+            }
+
+            for (int i = 0; i < routeEntities.size(); i ++) {
+                RouteEntry routeEntry = routeEntities.get(i);
+                String destination = routeEntry.getDestination();
+                String nexthop = routeEntry.getNexthop();
+                if (destination.equals(requestDestination) && nexthop.equals(requestNexthop)) {
+                    routeEntities.remove(i);
+                    break;
+                }
+            }
+        }
+        routeTable.setRouteEntities(routeEntities);
+        router.setRouteTable(routeTable);
+        this.routerDatabaseService.addRouter(router);
+
+        // Construct response
+        for (RouteEntry routeEntry : routeEntities) {
+            String destination = routeEntry.getDestination();
+            String nexthop = routeEntry.getNexthop();
+            RoutesToNeutronRouteObject routesToNeutronRouteObject = new RoutesToNeutronRouteObject(destination, nexthop);
+            responseRoutes.add(routesToNeutronRouteObject);
+        }
+        responseRouter.setId(routerid);
+        responseRouter.setName(router.getName());
+        responseRouter.setRoutes(responseRoutes);
+
+        return new RoutesToNeutronWebResponse(responseRouter);
+    }
+
+    @Override
+    public ConnectedSubnetsWebResponse getConnectedSubnets(String projectId, String vpcId, String subnetId) throws ResourceNotFoundException, ResourcePersistenceException, SubnetNotBindUniquePortId {
+        List<String> subnetIds = new ArrayList<>();
+
+        // get subnet
+        SubnetWebJson subnetWebJson = this.routerToSubnetService.getSubnet(projectId, subnetId);
+        SubnetEntity subnet = subnetWebJson.getSubnet();
+        if (subnet == null) {
+            logger.warn("can not find subnet by subnet id :" + subnetId);
+            return null;
+        }
+
+        // get subnet's route table type
+        String attachedRouterId = subnet.getAttachedRouterId();
+        if (attachedRouterId == null) {
+            return null;
+        }
+
+        Router router = this.routerDatabaseService.getByRouterId(attachedRouterId);
+        if (router == null) {
+            return null;
+        }
+        RouteTable routeTable = router.getRouteTable();
+        if (routeTable == null) {
+            return null;
+        }
+        RouteTableType routeTableType = routeTable.getRouteTableType();
+
+        if (routeTableType == null) {
+            return null;
+        } else if(routeTableType.getRouteTableType().equals("neutron")){
+            List<String> ports = router.getPorts();
+            // check ports
+            if (ports == null || ports.size() <= 1) {
+                return null;
+            }
+            for (String portId : ports) {
+                // get subnet by port id
+                SubnetsWebJson subnetsWebJson = this.routerToSubnetService.getSubnetsByPortId(projectId, portId);
+                if (subnetsWebJson == null) {
```

Comment:
@kevin-zhonghao We need to maintain subnet-port mapping in Route Manager instead of query the mapping from Subnet Manager.

Reply:
okay ~ left the TODO comments for next PR